pywhat/Data/regex.json

```diff
@@ -37,7 +37,7 @@
     },
     {
        "Name": "TryHackMe Flag Format",
-       "Regex": "(?i)thm{.*}|tryhackme{.*}",
+       "Regex": "(?i)^(thm{.*}|tryhackme{.*})$",

```

Comment:
This is a really nice way to avoid in-between regex! 👀

Reply:
There seem to be more regexes with in-between boundaries (URL, Coordinates, Discover Card, Visa, Solo, Switch).

`Amazon Web Services Simple Storage (AWS S3) URL` has in between boundaries and also seems to match to the whole line - it should at least stop if there is a space:
`(?i)^(?:https?:\/\/s3\.amazonaws\.com\/+[^\s]+|[a-z0-9_-]*\.s3\.amazonaws\.com\/[^\s]+)$`

Reply:
Would it not perhaps be easier to add a boundaryless regex: key to the data file?

Reply:
Maybe regexes should be changed to ^(pattern)$. Simple way to avoid problems with boundaries

Reply:
> Maybe regexes should be changed to ^(pattern)$. Simple way to avoid problems with boundaries

If we change all regexes to this format it should work yeah, and I don't think there is a need for a separate boundaryless key entry.